cloud/docker/docker_image.py

```diff
@@ -1,457 +1,502 @@
 #!/usr/bin/python
 #
-
-# (c) 2014, Pavel Antonov <antonov@adwz.ru>
+# Copyright 2016 Red Hat | Ansible
 #
 # This file is part of Ansible
 #
-# This module is free software: you can redistribute it and/or modify
+# Ansible is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation, either version 3 of the License, or
 # (at your option) any later version.
 #
-# This software is distributed in the hope that it will be useful,
+# Ansible is distributed in the hope that it will be useful,
 # but WITHOUT ANY WARRANTY; without even the implied warranty of
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU General Public License for more details.
 #
 # You should have received a copy of the GNU General Public License
-# along with this software.  If not, see <http://www.gnu.org/licenses/>.
-
-######################################################################
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 
 DOCUMENTATION = '''
 ---
 module: docker_image
-author: "Pavel Antonov (@softzilla)"
-version_added: "1.5"
-short_description: manage docker images
+
+short_description: Manage docker images.
+
+version_added: "2.1.0"
+
 description:
-     - Create, check and remove docker images
+     - Build, load or pull an image, making the image available for creating containers. Also supports tagging an
+       image into a repository and archiving an image to a .tar file.
+
 options:
-  path:
+  archive_path:
     description:
-       - Path to directory with Dockerfile
+      - Use with state 'present' to archive an image to a .tar file.
     required: false
     default: null
-    aliases: []
+  config_path:
+    description:
+      - Path to the Docker CLI config file.
+    default: '~/.docker/config.json'
```

Comment:
Actually config_path should be removed as path is defined further down. The correct thing is path, and path is the same as it was in the original module - the path to the docker build directory containing the Dockerfile.